jujugui/static/gui/src/app/components/basic-table/basic-table.js

```diff
@@ -0,0 +1,112 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/** Basic table React component used to display data in a table structure. */
+class BasicTable extends React.Component {
+  constructor() {
+    super();
+    this.columnClasses = [
+      'one-col',
+      'two-col',
+      'three-col',
+      'four-col',
+      'five-col',
+      'six-col',
+      'seven-col',
+      'eight-col',
+      'nine-col',
+      'ten-col',
+      'eleven-col',
+      'twelve-col'
+    ];
+  }
+
+  /**
+    Generate the a row.
+    @param header {Boolean} Whether this is a header row.
+    @param columnValues {Array} The list of column values.
+    @param index {Int} The row index (used for unique keys).
+    @returns {Object} The row to render.
+  */
+  _generateRow(header, columnValues, index) {
+    const columnsNumber = this.props.columns.length;
+    const columns = this.props.columns.map((column, i) => {
+      let conditionalClasses = {'last-col': i + 1 === columnsNumber};
+      // Map the column size to the appropriate CSS class.
+      conditionalClasses[this.columnClasses[parseInt(column.size) - 1]] = true;
+      if (column.classes) {
+        column.classes.forEach(className => {
+          conditionalClasses[className] = true;
+        });
+      }
+      const classes = classNames(conditionalClasses);
+      let content;
+      if (header) {
+        content = column.title;
+      } else {
+        content = columnValues[i] || '';
+      }
+      return (
+        <div className={classes}
```

Comment:
Typically we use a grid for tabular data as it can be modified to be displayed on a mobile whereas an actual `<table>` can't really be changed.